src/buffer.rs

```diff
@@ -119,6 +119,13 @@ impl<'a, P: Pixel + 'a> Iterator for Pixels<'a, P> where P::Subpixel: 'a {
     }
 }
 
+impl<'a, P: Pixel + 'a> ExactSizeIterator for Pixels<'a, P> where P::Subpixel: 'a {
```

Comment:
Can the entire where clause and all bounds on `P` be replaced with `where Self: Iterator`?